File: 백준/Silver/18113. 그르다 김가놈/그르다 김가놈.cc
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int N, K, M;
    vector<int> arr;
    cin >> N >> K >> M;
    
    for (int i=0; i<N; i++){
        int L;
        cin >> L;
        if (L<=K){ continue; }
        arr.push_back(L);
    }
    int lo = 1;
    int hi = 1000000000;
    int ans = -1;
    
    while (lo<=hi){
        int P = (lo+hi)/2;
        int cnt = 0;
        
        for(auto L : arr){
            cnt += L<(2*K) ? (L-K)/P : (L-(2*K))/P;
        }
        if (cnt >= M){
            ans = P;
            lo = P+1;
        }else{
            hi = P-1;
        }
    }
    cout << ans;
}

File: 백준/Silver/18113. 그르다 김가놈/그르다 김가놈.swift
import Foundation

let (N,K,M) = [readLine()!.split(separator: " ").map{Int($0)!}].map{($0[0], $0[1], $0[2])}[0]
var arr = [Int]()

for _ in 0..<N{
    let L = Int(readLine()!)!
    if L<=K{ continue }
    arr.append(L)
}
var lo = 1
var hi = 1000000000
var ans = -1

while lo<=hi{
    let P = (lo+hi)/2
    var cnt = 0
    
    for L in arr{
        cnt += L<2*K ? (L-K)/P:(L-(2*K))/P
    }
    
    if cnt >= M{
        ans = P
        lo = P+1
    }else{
        hi = P-1
    }
}
print(ans)